src/utils/asyncPopulate.js

```diff
@@ -14,6 +14,8 @@ export default function asyncPopulate(target, source) {
       promise = asyncPopulate(target[attr], source[attr]);
     } else if (source[attr] === null) {
       target[attr] = null;
+    } else if (typeof source[attr] === 'undefined') {
+      delete target[attr];
```

Comment:
Why do we delete this? I would've thought it'd be better to just pass through the undefined to the target.

Reply:
The primary motivation here is to provide developers with a way of telling factory-girl to leave a particular attribute of the built object out. Without this all attributes are always required.

Reply:
I'm against this in principle, unless there's a compelling reason why. Factories define a particular "type" of object, which should be stable at runtime. If you *do* need to omit a property at runtime then you can use `afterBuild` or some other hook to `delete` the property.

I'm still happy for this PR, because we want to avoid `getPrototypeOf(undefined)` throwing an exception, but I don't think we should `delete` any of the properties :+1:

Reply:
Hm, yeah, I see what you mean. That will probably work. I’ll give it a shot and report back.

Supposing passing through the value as `undefined` and then doing extra processing on it in `afterBuild` (in my case, deleting the key/value pair), would a diff like this be good to go?

```diff
+    } else if (typeof source[attr] === 'undefined') {
+      target[attr] = undefined;
```